src/Kernel/Class.class.st

```diff
@@ -885,7 +885,6 @@ Class >> removeFromSystem: logged [
 		self superclass addObsoleteSubclass: self].
 
 	myCategory := self category.
-	self isReferenced ifTrue: [Undeclared declare: self name asSymbol from: Smalltalk globals].
```

Comment:
The problem is what will happen when after removal we will load (create) class back? 
I did not check but I assume that methods pointing to old obsolete class will keep using it instead of new loaded class.

Reply:
There is no difference, as we did not recompile the methods (which would make it even slower).

Reply:
At least in my test it was still referencing the Obsolete class after re-loading. Just the syntax highlighter was happy and showing it as an existing class

Reply:
But this is a bug. Maybe I misunderstood.
Let's see an example: there is ClassA and there is a #methodWithClassA :
```Smalltalk
MyClassB class>>methodWithClassA 
  ^ClassA
```
"MyClassB methodWithClassA == ClassA" is true.
Now remove ClassA and create it again.
What will be the result of "MyClassB methodWithClassA == ClassA"? 
If it is not true it is a bug.

Reply:
Hmm... I tested it again and now it works.

Another idea: can we just add the class always to the Undeclared dict? Without the #isReferenced check?

This would accumulate entries, but we have that problem already. (e.g. imagine loading a method with an Undeclared and then delete it, the entry stays forever).

Reply:
I committed a version that does this